fuzzing/valuegeneration/mutator_shrinking.go

```diff
@@ -0,0 +1,207 @@
+	randomGeneratorDecision := g.randomProvider.Float32()
+	if randomGeneratorDecision < g.config.ShrinkValueProbability {
+		// Mutate the data for our desired number of rounds
+		input := bytesShrinkingMethods[g.randomProvider.Intn(len(bytesShrinkingMethods))](g, b)
+		return input
+	}
+	return b
+}
+
+// integerShrinkingMethods define methods which take a big integer and a set of inputs and
+// transform the integer with a random input and operation. This is used in a loop to create
+// mutated integer values.
+var integerShrinkingMethods = []func(*ShrinkingValueMutator, *big.Int, ...*big.Int) *big.Int{
+	func(g *ShrinkingValueMutator, x *big.Int, inputs ...*big.Int) *big.Int {
+		// Subtract a random input
+		var r *big.Int
+		if x.Cmp(big.NewInt(0)) > 0 {
+			r = big.NewInt(0).Sub(x, inputs[g.randomProvider.Intn(len(inputs))])
+		} else if x.Cmp(big.NewInt(0)) < 0 {
+			r = big.NewInt(0).Add(x, inputs[g.randomProvider.Intn(len(inputs))])
+		} else {
+			r = big.NewInt(0)
+		}
+		return r
+
+	},
+	func(g *ShrinkingValueMutator, x *big.Int, inputs ...*big.Int) *big.Int {
+		// Divide by two
+		return big.NewInt(0).Div(x, big.NewInt(2))
+	},
+	func(g *ShrinkingValueMutator, x *big.Int, inputs ...*big.Int) *big.Int {
+		return big.NewInt(0)
+	},
+}
+
+// MutateInteger takes an integer input and applies optional mutations to the provided value.
+// Returns an optionally mutated copy of the input.
+func (g *ShrinkingValueMutator) MutateInteger(i *big.Int, signed bool, bitLength int) *big.Int {
+	randomGeneratorDecision := g.randomProvider.Float32()
+	if randomGeneratorDecision < g.config.ShrinkValueProbability {
+		// Calculate our integer bounds
+		min, max := utils.GetIntegerConstraints(signed, bitLength)
+
+		// Obtain our inputs. We also add our min/max values for this range to the list of inputs.
+		// Note: We exclude min being added if we're requesting an unsigned integer, as zero is already
+		// in our set, and we don't want duplicates.
+		var inputs []*big.Int
+		inputs = append(inputs, g.valueSet.Integers()...)
+		if signed {
+			inputs = append(inputs, min, max)
+		} else {
+			inputs = append(inputs, max)
+		}
+
+		// Determine which value we'll use as an initial input, and how many mutations we will perform.
+		inputIdx := g.randomProvider.Intn(len(inputs))
+		input := new(big.Int)
+		if i != nil {
+			input.Set(i)
+		} else {
+			input.Set(inputs[inputIdx])
+		}
+		input = utils.ConstrainIntegerToBounds(input, min, max)
+
+		// Shrink input
+		input = integerShrinkingMethods[g.randomProvider.Intn(len(integerShrinkingMethods))](g, input, inputs...)
+
+		// Correct value boundaries (underflow/overflow)
+		input = utils.ConstrainIntegerToBounds(input, min, max)
+		return input
+	}
+	return i
+}
+
+// stringShrinkingMethods define methods which take an initial string and a set of inputs to transform the input. The
+// transformed input is returned. This is used in a loop to mutate strings.
```

Comment:
Is the function used in a loop? This pertains to the last sentence of the comment

Reply:
https://github.com/crytic/medusa/pull/147/commits/dd31ac5546a8dd1f41e54d03157e1165213fcbd8